exchanges/stream/websocket.go

```diff
@@ -410,15 +573,21 @@ func (w *Websocket) connectionMonitor() error {
 			}
 			select {
 			case err := <-w.ReadMessageErrors:
-				w.DataHandler <- err
-				if IsDisconnectionError(err) {
+				if errors.Is(err, errConnectionFault) {
 					log.Warnf(log.WebsocketMgr, "%v websocket has been disconnected. Reason: %v", w.exchangeName, err)
 					if w.IsConnected() {
 						if shutdownErr := w.Shutdown(); shutdownErr != nil {
 							log.Errorf(log.WebsocketMgr, "%v websocket: connectionMonitor shutdown err: %s", w.exchangeName, shutdownErr)
 						}
 					}
 				}
+				// Speedier reconnection, instead of waiting for the next cycle.
+				if w.IsEnabled() && (!w.IsConnected() && !w.IsConnecting()) {
+					if connectErr := w.Connect(); connectErr != nil {
+						log.Errorln(log.WebsocketMgr, connectErr)
+					}
+				}
+				w.DataHandler <- err // hand over the error to the data handler (shutdown and reconnection is priority)
```

Comment:
The test issue could be from here. The FlushChannels calls `Shutdown`, should a connectionmonitor restart a connection from there?